### PR TITLE
Momentum and Learning Rate Decay for SGD Optimizer

### PR DESCRIPTION
**Pull request by kabirkhan**
# Description

Adds support for optional (default to 0.0) momentum and learning rate decay for the Stochastic Gradient Descent SGD optimizer. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Added tests for an existing feature

# How Has This Been Tested?

Run the notebook: notebooks/Openmined/notebooks/demos/MLP Using Optimizer.ipynb
Remove momentum and decay to validate that the loss is the same as before

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by iamtrask** [comment]
Dude I'm so excited about your work here!!! Please keep going!!! Any chance you'd be interested in RMSProp, Adam, or Adagrad?
